util.ts

```diff
@@ -0,0 +1,44 @@
+import zlib = require('zlib');
+import {IHttpClientResponse} from './interfaces'
+
+/**
+ * Decompress/Decode gzip encoded JSON
+ * Using Node.js built-in zlib module
+ *
+ * @param {Buffer} buffer
+ * @param {string} charset
+ * @return {Promise<string>}
+ */
+export function decompressGzippedContent(buffer: Buffer, charset: string): Promise<string> {
+    return new Promise<string>((resolve, reject) => {
+        zlib.gunzip(buffer, function (error, buffer) {
+            if (error) {
+                reject(error);
+            }
+
+            resolve(buffer.toString(charset));
+        });
+    })
+}
+
+/**
+ * Obtain Response's Content Charset.
+ * Through inspecting `content-type` response header.
+ * It Returns 'utf-8' if NO charset specified/matched.
+ *
+ * @param {IHttpClientResponse} response
+ * @return {string} - Content Encoding Charset; Default=utf-8
+ */
+export function obtainContentCharset (response: IHttpClientResponse) : string {
```

Comment:
I might be missing something, but I think it would be better for this to return `binary` for [`Content-type: application/octet-stream`](https://github.com/konradpabjan/artifact-test/runs/549444862?check_suite_focus=true#step:5:66).

My repl test - using `binary` preserves my original three bytes:

```
> Buffer.from(Buffer.from('ffffff', 'hex').toString('binary'), 'binary')
<Buffer ff ff ff>

> Buffer.from(Buffer.from('ffffff', 'hex').toString('utf-8'), 'utf-8')
<Buffer ef bf bd ef bf bd ef bf bd>
```